### PR TITLE
Reconcile

### LINKED ISSUE #57: A light's zone ID is not changed if the zone is deleted
The zone ID on a light should be set to `null` when a zone is deleted

### PR DESCRIPTION
**Pull request by nebrius**
This fills in the missing gaps where updating one part of state wasn't being reflected in other parts of state, such as moving a light from one zone to another needing to remove the light entry from scenes in old zone and adding it to scenes in the new zone.

Fixes #57 and #62

### CODE REVIEW

**Comment by nebrius** [comment]
I went ahead and added null coalescing to fix #59 as part of this PR, since it touched on a lot of the same code

File: server/src/db/philipsHue.ts [hunk 1]
```diff
@@ -31,11 +31,11 @@ export interface PhilipsHueInfo {
   key: string;
 }
 
-export async function getPhilipsHueInfo(): Promise<PhilipsHueInfo | null> {
+export async function getPhilipsHueInfo(): Promise<PhilipsHueInfo | undefined> {
   const rows = await dbAll(`SELECT * FROM ${PHILIPS_HUE_TABLE_NAME}`);
   switch (rows.length) {
     case 0:
-      return null;
+      return undefined;
```
Comment (allxie) [inline server/src/db/philipsHue.ts:38]: doesn't "return" just return undefined?

File: server/src/db/scenes.ts [hunk 2]
```diff
@@ -48,6 +55,74 @@ export async function init(): Promise<void> {
...
+      ]);
+      changesMade = true;
+    }
+  }
+
+  // Update the cache if changes were made
+  if (changesMade) {
+    await updateCache();
+  }
+  console.log(zones, patterns, lights);
```
Comment (allxie) [inline server/src/db/scenes.ts:123]: Did you mean to leave this?

File: server/src/db/philipsHue.ts (hunk 1 above)
Comment (nebrius) [reply server/src/db/philipsHue.ts:38]: Haha, oh right, will change.

File: server/src/db/scenes.ts (hunk 2 above)
Comment (nebrius) [reply server/src/db/scenes.ts:123]: Nope, will remove

File: server/src/db/philipsHue.ts (hunk 1 above)
Comment (nebrius) [reply server/src/db/philipsHue.ts]: Fixed

File: server/src/db/scenes.ts (hunk 2 above)
Comment (nebrius) [reply server/src/db/scenes.ts]: Fixed

File: server/src/device/phillipsHue.ts [hunk 3]
```diff
@@ -122,19 +122,19 @@ async function discoverBridge(): Promise<string | null> {
     console.log(
       'Failed to search for Philips Hue bridges, calls to set state on Philips Hue lights will be ignored'
     );
-    return null;
+    return undefined;
```
Comment (allxie) [inline server/src/device/phillipsHue.ts:125]: Also these, please

**Review by allxie — CHANGES_REQUESTED** [review]

File: server/src/device/phillipsHue.ts (hunk 3 above)
Comment (nebrius) [reply server/src/device/phillipsHue.ts]: Done

**Review by allxie — APPROVED** [review]

**Pull request merged**